### ISSUE TITLE
CPU support

### ISSUE THREAD

**Issue by backplanellc** [brichard19/BitCrack#71]
Any plans for CPU support? 

This works great thank you. I had to do some tweaks in Makefile for my gpu architecture but that was it. 

I get roughly 50mkey/s on a 1070 gtx with 600mb address file. 

Thanks again for your great work.

**Comment by bill32767** [comment]
You should get more that 180mkeys/s on your card. Increase the value of -b
-t -p (for instance -b 64 -t 256 -p 1024). You can keep playing around with
the values of -t -p until you reach a maximum.

**Comment by brichard19** [comment]
See #24 

I'd like to do a CPU version since AMD is now offering 32-core chips.

The priority right now is GPU. OpenCL does not perform very well so we are looking I to ROCm ti get the AMD GPU performance up to par with Nvidia.

And bill is right, you should probably be getting more than 50 million keys per second on a GTX 1070.

**Comment by bill32767** [comment]
1080 gtx card gets about 343mkeys/s for compressed address and a little
lesser for uncomprosed address

**Comment by backplanellc** [comment]
After adjusting those settings as advised, i am able to get about 300MKey/s!

I'm dying to use my 20 core/40 thread system with this so i anxiously await any CPU support. I'd even be willing to donate to the cause.

-Dennis

**Comment by Dragoula** [comment]
You get 300Mkeys/s with how many addresses in the file ?

**Comment by Granddadi** [comment]
Please write me a mail ylsanlydze@gmail.com. I have some solution and a script that may change the performance. In a nutshell I can not explain.

**Comment by Coinpiet** [comment]
I have 200k addresses plus the 160 puzzle addresses and I have 5 gtx 1070 cards. But I have to open a new terminal for every card and I only get 23 mkeys per card what can I do I use the ndv oclvanitygen

**Comment by ogronome** [comment]
oclvanitygen from ndv is twice slower then bitcrack. I funded NDV's work but not using it any more. I wish I've donated to brichard :( 

on single gtx1070 I get 245MK with 4k list and only compressed addresses
-b 15 -t 512 -p 4096

there is a recommendation by bill 
-b 64 -t 256 -p 1024
but it makes only 176MKps for me.

**Comment by bill32767** [comment]
Yes that was an example I gave for gtx 1080i. When you reduce -b and
increase -t and -p the number of keys/sec increases. I think your gtx1070
can do more than 245MK, just slightly increase -t and -p (with no. 8 or 16
or 32 or 64) until it reaches its limit.

**Comment by Coinpiet** [comment]
wow ok so i do now with my 5 GTX1070 190 Mkeys with vanitygen 
can Bitcrack do both like LBC uncompressed and compressed ?
can someone help me to start the Bitcrack ?

**Comment by Coinpiet** [comment]
and i have a 42k list to search and i let every card run in another space

**Comment by backplanellc** [comment]
How can we donate to your cause brichard19 :)

**Comment by reppolice** [comment]
@backplanellc what were your settings? I only managed 90M with variations of the suggested parameters above
